diff --git a/chrome/browser/profiles/profile_impl.cc b/chrome/browser/profiles/profile_impl.cc
index 71619ca3d5271b0f813110dce0e5570b6eebdda7..ab74c9c61fd94604e94fc07df0d52968932093b2 100644
--- a/chrome/browser/profiles/profile_impl.cc
+++ b/chrome/browser/profiles/profile_impl.cc
@@ -363,7 +363,7 @@ std::unique_ptr<Profile> Profile::CreateProfile(const base::FilePath& path,
   }
 
   std::unique_ptr<Profile> profile = base::WrapUnique(
-      new ProfileImpl(path, delegate, create_mode, io_task_runner));
+      new BraveProfileImpl(path, delegate, create_mode, io_task_runner));
 #if BUILDFLAG(ENABLE_SUPERVISED_USERS) && !defined(OS_ANDROID) && \
     !defined(OS_CHROMEOS)
   if (create_mode == CREATE_MODE_SYNCHRONOUS && profile->IsLegacySupervised())
@@ -508,6 +508,9 @@ ProfileImpl::ProfileImpl(
   LoadPrefsForNormalStartup(async_prefs);
 #endif
 
+// Done here instead of browser_prefs.cc so it can override default prefs
+brave::RegisterProfilePrefs(pref_registry_.get());
+
   content::BrowserContext::Initialize(this, path_);
 
   // Register on BrowserContext.
