diff --git a/chrome/browser/profiles/profile_impl.h b/chrome/browser/profiles/profile_impl.h
index 28315594f50c07150a1d29a5546cc7359a93cbb2..b9fb33e287951b8723364df8efcccd8c704e0afe 100644
--- a/chrome/browser/profiles/profile_impl.h
+++ b/chrome/browser/profiles/profile_impl.h
@@ -179,6 +179,7 @@ class ProfileImpl : public Profile {
 #endif  // defined(OS_CHROMEOS)
 
  private:
+  friend class BraveProfileImpl;
 #if defined(OS_CHROMEOS)
   friend class chromeos::KioskTest;
   friend class chromeos::SupervisedUserTestBase;
