diff --git a/extensions/browser/api/web_request/web_request_api.cc b/extensions/browser/api/web_request/web_request_api.cc
index b83d047e7fdb2391dd74cfd30c8b003455e8e8c3..d8827c15268e11c149365ea16e690864ea52d0f0 100644
--- a/extensions/browser/api/web_request/web_request_api.cc
+++ b/extensions/browser/api/web_request/web_request_api.cc
@@ -747,6 +747,7 @@ void WebRequestAPI::ProxyWebSocket(
           frame->GetProcess()->GetBrowserContext());
 
   WebRequestProxyingWebSocket::StartProxying(
+      frame->GetFrameTreeNodeId(),
       std::move(factory), url, site_for_cookies, user_agent,
       handshake_client.PassInterface(), has_extra_headers,
       frame->GetProcess()->GetID(), frame->GetRoutingID(),
@@ -1243,6 +1244,9 @@ ExtensionWebRequestEventRouter::OnAuthRequired(
     const net::AuthChallengeInfo& auth_info,
     net::NetworkDelegate::AuthCallback callback,
     net::AuthCredentials* credentials) {
+  ClearSignaled(request->id, kOnBeforeSendHeaders);
+  ClearSignaled(request->id, kOnSendHeaders);
+  ClearSignaled(request->id, kOnHeadersReceived);
   // No browser_context means that this is for authentication challenges in the
   // system context. Skip in that case. Also skip sensitive requests.
   if (!browser_context ||
